src/components/toggle/toggle.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import {findDOMNode} from 'react-dom';
+import {SBComponent} from 'stylable-react-component';
+import style from './toggle.st.css';
+
+export interface Props {
+    checked?: boolean,
+    error?: boolean,
+    disabled?: boolean,
+    onChange?: (selected: boolean) => void,
+    label?: string,
+    tabIndex?: number,
+    rtl?: boolean
+}
+export interface State {
+    focus: boolean,
+    click: boolean
+}
+
+@SBComponent(style)
+export default class Toggle extends React.Component<Props, State> {
+    static defaultProps = {
+        checked: false,
+        disabled: false,
+        error: false,
+        rtl: false
+    }
+    state = {
+        focus: false,
+        click: false
+    }
+    toggle = (e: React.SyntheticEvent<HTMLInputElement>) => {
+        if (!this.props.disabled && this.props.onChange) {
+            this.props.onChange(!this.props.checked)
+        }
+        // reseting focus when user click toggle
+        if (this.state.click) {
+            this.setState({
+                click: false,
+                focus: false
+            })
+        }
+    }
+    render() {
+        const {
+            checked,
+            disabled,
+            error,
+            rtl,
+            label,
+            tabIndex,
+            onChange
+        } = this.props;
+        const {focus} = this.state;
+
+        return <label
+            data-automation-id='TOGGLE'
+            onMouseDown={() => this.setState({click: true})}
```

Comment:
I guess spec is about public interface of component, not the possible listeners for inner child. For this specific case I have to lister for `mousedown` not for `click`. The reason I need this is that I need to reset `focus` state for component when user clicked on `Toggle` and do not touch `focus` when user change value with `space`/`enter` when focused on component.
Current behaviour of native input is tiggering `click` event when user use `space`/`enter` and this click bubbles to `Toggle`, but this will not happend with `mousedown` event